proto/penumbra/penumbra/view/v1alpha1/view.proto

```diff
@@ -115,8 +134,20 @@ message AuthorizeAndBuildRequest {
 }
 
 message AuthorizeAndBuildResponse {
-  // The transaction that was built.
-  core.transaction.v1alpha1.Transaction transaction = 1;
+  // Signals that building is in progress.
+  message BuildProgress {
+    // An approximate progress of the build, from 0 to 1.
+    float progress = 1;
+  }
+  // Signals that the transaction is complete.
+  message Complete {
+    // The finished transaction.
+    penumbra.core.transaction.v1alpha1.Transaction transaction = 1;
+  }
+  oneof status {
+    BuildProgress build_progress = 1;
+    Complete complete = 2;
+  }
```

Comment:
Looks like these can be unnested so it can be shared with WitnessAndBuildResponse too

Reply:
Sounds good, feel free to pick it up, we'll also have to propagate these changes through the rest of the source code.